lib/cli/ui/prompt/options_handler.rb

```diff
@@ -15,8 +15,16 @@ def option(option, &handler)
           @options[option] = handler
         end
 
-        def call(option)
-          @options[option].call(option)
+        def call(options)
+          case options
+          when Array
+            # Slice out the handlers and call each with the option they're for
+            @options.values_at(*options).each_with_index.map do |handler, index|
```

Comment:
This seems overly complex.  Am I missing something? Is there a reason we can't
```ruby
Array(options).each do |option|
  @options[option].call(option)
end
```
as the entire method body?

Reply:
That would force ask to always return an array of values, which is not how it currently behaves.

Aside from that, uh, yeah, I'm not sure why I didn't just map over the options.

Reply:
I’m not sure I follow.

Note that `Array(foo)` will convert to array only if not already an array:
```rb
irb(main):004:0> Array(1)
=> [1]
irb(main):005:0> Array([1])
=> [1]
```
I see @lugray's suggestion as being simpler and equivalent.

Reply:
Because, that changes the response when only giving a single answer.

Reply:
@lugray @lavoiesl I think it's better if I just show you what I'm trying to say.

Current implementation:
```ruby
2.5.1 :011 > CLI::UI::Prompt.ask('q') do |handler|
2.5.1 :012 >       ('1'..'10').each do |i|
2.5.1 :013 >           handler.option(i) { i }
2.5.1 :014?>       end
2.5.1 :015?>   end
? q (You chose: 5)
 => "5"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
2.5.1 :016 > CLI::UI::Prompt.ask('q', multiple: true) do |handler|                                                                                                                                                                                                                                                                                                                                                                                                                             
2.5.1 :017 >       ('1'..'10').each do |i|                                                                                                                                                                                                                                                                                                                                                                                                                                                     
2.5.1 :018 >           handler.option(i) { i }                                                                                                                                                                                                                                                                                                                                                                                                                                                 
2.5.1 :019?>       end                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
2.5.1 :020?>   end                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
? q (You chose: 3 items)
 => ["2", "5", "8"]
```

With `OptionsHandler#call` as above:

```ruby
2.5.1 :003 > CLI::UI::Prompt.ask('q') do |handler|
2.5.1 :004 >       ('1'..'10').each do |i|
2.5.1 :005 >           handler.option(i) { i }
2.5.1 :006?>       end
2.5.1 :007?>   end
? q (You chose: 1)
 => ["1"]    # <----- this is now different                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
2.5.1 :008 >                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
2.5.1 :009 > CLI::UI::Prompt.ask('q', multiple: true) do |handler|                                                                                                                                                                                                                                                                                                                                                                                                                             
2.5.1 :010 >       ('1'..'10').each do |i|                                                                                                                                                                                                                                                                                                                                                                                                                                                     
2.5.1 :011 >           handler.option(i) { i }                                                                                                                                                                                                                                                                                                                                                                                                                                                 
2.5.1 :012?>       end                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
2.5.1 :013?>   end                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
? q (You chose: 3 items)
 => ["1", "4", "8"] 
```

Reply:
Maybe I'm the one who's not understanding.

@lugray, by "as the entire method body" I thought you meant the `call` method should be:

```ruby
def call(options)
  Array(options).each do |option|
    @options[option].call(option)
  end
end
```

Is that not what you meant?

Reply:
Ah! I see now, I missed that the return value was important.

Thanks for the explanation.

Reply:
Ah!  That was my suggestion, but to preserve the return value.:
```ruby
def call(options)
  Array(options).each do |option|
    @options[option].call(option)
  end
  options
end
```

Reply:
That would be returning the original `options` argument, not the return value of `call`